python/ray/tests/test_top_level_api.py

```diff
@@ -0,0 +1,45 @@
+from inspect import getmembers, isfunction
+
+import ray
+
+
+def test_api_functions():
+    # These are pulled from the "Python API" of the Ray Docs
+    # util.queue.Queue & util.ActorPool are removed because they are under
+    # util.*, not the top-level Ray namespace.
+    PYTHON_API = [
+        "init",
+        "is_initialized",
+        "remote",
+        "get",
+        "wait",
+        "put",
+        "kill",
+        "cancel",
+        "get_gpu_ids",
+        "shutdown",
+        "method",
+        "nodes",
+        "timeline",
+        "cluster_resources",
+        "available_resources",
+        "java_function",
+        "java_actor_class",
+    ]
+
+    OTHER_ALLOWED_FUNCTIONS = [
+        # TODO(ilr) either move this to PYTHON_API or remove.
+        "get_actor",
+        # In the RuntimeContext API
+        "get_runtime_context",
+    ]
+
+    functions = getmembers(ray, isfunction)
+    function_names = [f[0] for f in functions]
+    assert set(function_names) == set(PYTHON_API + OTHER_ALLOWED_FUNCTIONS)
+
```

Comment:
@AmeerHajAli Here is the test!

Reply:
This is really good. Lets add a comment for why we are testing this and making it clear that adding 1 more public API requires supporting it in Ray Client.

Reply:
Sure, I'll mention that here. This did make me wonder, should we add client support for the Cross Language API?